hpat/tests/test_series.py

```diff
@@ -2104,7 +2179,84 @@ def test_impl():
             return S.nsmallest(4)
         hpat_func = hpat.jit(test_impl)
 
-        np.testing.assert_array_equal(hpat_func().values, test_impl().values)
+        np.testing.assert_array_equal(hpat_func(), test_impl())
+        self.assertEqual(count_parfor_REPs(), 0)
+        self.assertTrue(count_array_OneDs() > 0)
+
+    @unittest.skipIf(hpat.config.config_pipeline_hpat_default,
+                     'Series.nsmallest() parameter keep unsupported')
+    def test_series_nsmallest_full(self):
+        def test_impl(series, n, keep):
+            return series.nsmallest(n=n, keep=keep)
+        hpat_func = hpat.jit(test_impl)
+
+        keep = 'first'
+        for data in test_global_input_data_numeric + [[]]:
+            series = pd.Series(data * 3)
+            for n in range(-1, 10):
+                ref_result = test_impl(series, n, keep)
+                jit_result = hpat_func(series, n, keep)
+                pd.testing.assert_series_equal(ref_result, jit_result)
+
+    @unittest.skipIf(hpat.config.config_pipeline_hpat_default,
+                     'Series.nsmallest() index unsupported')
+    def test_series_nsmallest_index(self):
+        def test_impl(series, n):
+            return series.nsmallest(n=n)
+        hpat_func = hpat.jit(test_impl)
+
+        # TODO: check data == [] when index fixed
+        for data in test_global_input_data_numeric:
+            data *= 3
+            for index in [gen_srand_array(len(data)), range(len(data))]:
+                series = pd.Series(data, index)
+                for n in range(-1, 10):
+                    ref_result = test_impl(series, n)
+                    jit_result = hpat_func(series, n)
+                    pd.testing.assert_series_equal(ref_result, jit_result)
```

Comment:
you can use `np.testing.assert_array_equal(ref_result.value, jit_result.value)` to check for "old style"

Reply:
I see this test is also dosn't work with index of strings. Also, as far s i remember, old-style is working with index of strings.
Anyway, you could use if/else as you did in other tests to make diffrent checks for diffrent styles. Please note, I proposed `ref_result.value` which returns array (not series)